core/src/main/scala/tapir/generic/SchemaForMagnoliaDerivation.scala

```diff
@@ -53,7 +52,7 @@ trait SchemaForMagnoliaDerivation {
   }
 
   def dispatch[T](ctx: SealedTrait[SchemaFor, T]): SchemaFor[T] = {
-    throw new RuntimeException(s"Sealed trait hierarchies are not yet supported for: ${ctx.typeName}")
+    SchemaFor(SCoproduct(ctx.subtypes.map(_.typeclass.schema).toList, None))
```

Comment:
why `None`? is a coproduct schema useful in any way without a discriminator?

Reply:
Yes, the discriminator according to openApi is optional.
```
In this case, a discriminator MAY act as a "hint" to shortcut validation and selection of the matching schema which may be a costly operation, depending on the complexity of the schema. We can then describe exactly which field tells us which schema to use:
```